src/main/java/com/useful/ucars/uCarsListener.java

```diff
@@ -879,7 +881,7 @@ public void onUcarUpdate(ucarUpdateEvent event) {
 			}
 		}
 		
-		Vector travel = event.getTravelVector(); // Travel Vector,
+		Vector travel = event.getTravelVector().setY(event.getVehicle().getVelocity().getY()); // Travel Vector,
```

Comment:
Hmm, are you sure that this doesn't break things with block climbing and other situations where the car velocity is modified un-naturally? This also overwrites the y value calculated in "Motion Manager" (Yes this plugin's code really is a mess..). I suspect that with this conservation of y velocity after a block-climb or similar which perturbs the velocity momentarily to a large value the car may want to shoot up into the air. (There was an old bug that used to be that if you block-climbed at the same time as stopping pressing the accelerate button the car would shoot upwards as the y velocity would be conserved. Sometimes quite a few blocks into the air...)

Reply:
That's interesting, I'll try to reproduce that.
The problem I was trying to address with these changes was that cars could not be thrown upwards (more than one block) by stuff like bombs in the MarioKart-Plugin after the last changes.

I'll look into it

Reply:
I think there is an existing way to denote the car is being flung, but see what you can find I guess. Maybe not car.penalty but some other name that denotes the y velocity is conserved instead as it's more meaningful :P

Reply:
What about "car.decoupleY"?

Reply:
or car.inertialYAxis? or something like that?